Code/src/main/java/nl/utwente/group10/haskell/expr/Function.java

```diff
@@ -60,8 +60,12 @@ public String toString() {
      * @param expr The expression that forms the base of this function.
      */
     public Function(Expr expr, FunctionArgument ... arguments) {
-        this.arguments = Arrays.asList(arguments);
+        this(expr, Arrays.asList(arguments));
+    }
+
+    public Function(Expr expr, List<FunctionArgument> arguments) {
```

Comment:
Could you copy the javadoc to this constructor as well? That way it will be consistent throughout the back end.